Exiled.API/Features/Roles/Role.cs

```diff
@@ -0,0 +1,148 @@
+        /// </summary>
+        public UnityEngine.Color Color => Type.GetColor();
+
+        /// <summary>
+        /// Gets a value indicating whether or not this role is still valid. This will only ever be <see langword="false"/> if the Role is stored and accessed at a later date.
+        /// </summary>
+        public bool IsValid => Type == Owner.ReferenceHub.characterClassManager.NetworkCurClass;
+
+        /// <summary>
+        /// Gets the RoleType belonging to this role.
+        /// </summary>
+        internal abstract RoleType RoleType { get; }
+
+        /// <summary>
+        /// Converts a role to its appropriate <see cref="global::RoleType"/>.
+        /// </summary>
+        /// <param name="role">The role.</param>
+        public static implicit operator RoleType(Role role) => role.Type;
+
+        /// <summary>
+        /// Returns whether or not the role has the same RoleType as the given <paramref name="type"/>.
+        /// </summary>
+        /// <param name="role">The role.</param>
+        /// <param name="type">The <see cref="global::RoleType"/>.</param>
+        /// <returns><see langword="true"/> if the values are equal.</returns>
+        public static bool operator ==(Role role, RoleType type)
+            => role.Type == type;
+
+        /// <summary>
+        /// Returns whether or not the role has a different RoleType as the given <paramref name="type"/>.
+        /// </summary>
+        /// <param name="role">The role.</param>
+        /// <param name="type">The <see cref="global::RoleType"/>.</param>
+        /// <returns><see langword="true"/> if the values are not equal.</returns>
+        public static bool operator !=(Role role, RoleType type)
+            => role.Type != type;
+
+        /// <summary>
+        /// Returns whether or not the role has the same RoleType as the given <paramref name="type"/>.
+        /// </summary>
+        /// <param name="type">The <see cref="global::RoleType"/>.</param>
+        /// <param name="role">The role.</param>
+        /// <returns><see langword="true"/> if the values are equal.</returns>
+        public static bool operator ==(RoleType type, Role role)
+            => role.Type == type;
+
+        /// <summary>
+        /// Returns whether or not the role has a different RoleType as the given <paramref name="type"/>.
+        /// </summary>
+        /// <param name="type">The <see cref="global::RoleType"/>.</param>
+        /// <param name="role">The role.</param>
+        /// <returns><see langword="true"/> if the values are not equal.</returns>
+        public static bool operator !=(RoleType type, Role role)
+            => role.Type != type;
+
+        /// <summary>
+        /// Casts the role to the specified role type.
+        /// </summary>
+        /// <typeparam name="T">The type of the class.</typeparam>
+        /// <returns>The casted class, if possible.</returns>
+        public T As<T>()
```

Comment:
Why don't you make a safe `As<T>` version? Basically it would be `bool As<T>(out T param)`, which is better than doing `Role::As<T> != null`.

Reply:
sounds hot